common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -256,6 +267,48 @@ def has_support(self, view, functionality):
             )
         return False
 
+    def validate(self):
+        """
+        Validates the xml for correct question separation. There must be only one inputtype in every <question>.
+        This is the override of the general XBlock method.
+        """
+
+        validation = super(CapaDescriptor, self).validate()
+        validation = StudioValidation.copy(validation)
+
+        # validate xml to check questions are correctly separated
+        is_valid = self.validate_xml()
+
+        if not is_valid:
+            validation.set_summary(
+                StudioValidationMessage(
+                    StudioValidationMessage.WARNING,
+                    u'Incorrect problem format detected!',
+                    action_label=u"Please see docs.",
+                    action_class=u"external-link",
+                    action_link=u"http://docs.edx.org/"
+                )
+            )
+
+        return validation
+
+    def validate_xml(self):
+        input_tags = ['choiceresponse', 'optionresponse', 'multiplechoiceresponse', 'stringresponse',
+                      'numericalresponse', 'schematicresponse', 'imageresponse', 'formularesponse']
+
```

Comment:
@ehteshamkafeel We should avoid to hardcode the inputtypes instead try something like

```Python
import capa.inputtypes as inputtypes
inputtypes = inputtypes.registry.registered_tags()
```